apps/api/src/domain/models/categories/entities/category.validateExternalState.spec.ts

```diff
@@ -0,0 +1,128 @@
+import { InternalError } from '../../../../lib/errors/InternalError';
+import buildTestData from '../../../../test-data/buildTestData';
+import { DeepPartial } from '../../../../types/DeepPartial';
+import { Valid } from '../../../domainModelValidators/Valid';
+import { InMemorySnapshot, resourceTypes } from '../../../types/resourceTypes';
+import { Term } from '../../term/entities/term.entity';
+import InvalidExternalReferenceInCategoryError from '../errors/InvalidExternalReferenceInCategoryError';
+import { noteType } from '../types/ResourceTypeOrNoteType';
+import { Category } from './category.entity';
+
+const missingTerm = buildTestData().resources.term[0].clone<Term>({
+    id: 'id-of-missing-term-id',
+});
+
+const validCategory = new Category({
```

Comment:
This is dangerous. You should build the DTO and type it separately, since the constructor trusts its input. It is meant to only be called within a factory that validates invariants.

Reply:
fixed